src/huggingface_hub/commit_api.py

```diff
@@ -0,0 +1,424 @@
+"""
```

Comment:
if we have this file as `_commit_api.py`, we won't need to deal with backward compatible changes if we move methods around which users shouldn't be using.